src/puppetlabs/puppetdb/queue.clj

```diff
@@ -147,32 +160,62 @@
    (let [rx (metadata-rx (keys metadata-command->puppetdb-command))
          md-cmd->pdb-cmd metadata-command->puppetdb-command]
      (fn [s]
-       (when-let [[_ stamp md-command version certname] (re-matches rx s)]
-         (and certname
-              {:received (-> stamp
-                             Long/parseLong
-                             tcoerce/from-long
-                             kitchensink/timestamp)
-               :version (Long/parseLong version)
-               :command (get md-cmd->pdb-cmd md-command "unknown")
-               :certname certname}))))))
+       (when-let [[_ received-stamp producer-offset md-command version certname] (re-matches rx s)]
+         (let [received-time-long (Long/parseLong received-stamp)
+               producer-offset (and producer-offset (Long/parseLong producer-offset))]
+           (and certname
```

Comment:
I don't remember the scenarios, @rbrw might have more info. Garbage in the directory maybe? I think this is used by both the dead letter code and the queue code